lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -4818,7 +4818,8 @@ namespace Js
 #endif
         }
 
-        if (result == FALSE && (propertyOperationFlags & PropertyOperation_StrictMode))
+        if (result == FALSE && 
+            (propertyOperationFlags & (PropertyOperation_StrictMode | PropertyOperation_ThrowOnDeleteIfNotConfig)))
         {
             JavascriptError::ThrowCantDelete(propertyOperationFlags, scriptContext, GetPropertyDisplayNameForError(index, scriptContext)->GetString());
```

Comment:
Doesn't `JavascriptError::ThrowCantDelete` already check `propertyOperationFlags`? Can we write this as:

```cpp
if (!result)
{
    JavascriptError::ThrowCantDelete(propertyOperationFlags, scriptContext, GetPropertyDisplayNameForError(index, scriptContext)->GetString());
}
```